lib/rex/tar.rb

```diff
@@ -0,0 +1,8 @@
+# -*- coding: binary -*-
+
+require 'rubygems/package'
+
```

Comment:
I would leave documentation for posterity here, or in the class definition, explaining that changes to standard functionality such as creating broken archives or decompressing them in some incorrect but required manner goes into these namespaces, with appropriate method names to the function, etc.

Reply:
Yeah, I don't want to leave us with more to refactor later. This is very much WIP. Let's design this out right from the start.

Reply:
Currently toying with the idea of `GenericReader` and `GenericWriter`. And if we do decide to combine everything into `Rex::Archive`, what will we do then? Don't want to create more work for us down the road.

Reply:
GenericFormatDiviner...